components/cam/bld/config_files/definition.xml

```diff
@@ -319,6 +318,13 @@ that setting to allow for cross-compilation, and for instances where the
 $OSNAME value is too generic.  For example, currently on both cray-xt and
 bluegene systems $OSNAME has the value "linux".
 </entry>
+<entry id="mach" value="">
```

Comment:
So does this need to be set separate from the "mach" value that goes into the create_new_case command?
BTW, Summit is another problematic machine in this respect. The value for hostname is like "login3".

Reply:
I put this is because similar logic is in place for identifying the OS (right above or below this), and something needs to be specified. In practice, the mach variable from create_newcase is used. See cam/cime_config/config_component.xml, where I added

           <value compset=""               >-mach $MACH</value>

This shows up as the first option in the CAM_CONFIG_OPTS line in env_build.xml.